File: frontend/src/api/crash.ts
import { api } from "./client";

export interface CrashReportRow {
  id: string;
  created_at: string;
  app_version: string | null;
  android_version: string | null;
  device: string | null;
  installation_id: string | null;
  stack_trace: string | null;
  user_comment: string | null;
}

export interface CrashReportList {
  items: CrashReportRow[];
  limit: number;
  offset: number;
  has_more: boolean;
}

export async function listCrashReports(limit = 50, offset = 0): Promise<CrashReportList> {
  const { data } = await api.get<CrashReportList>("/admin/crash-reports", {
    params: { limit, offset },
  });
  return data;
}

File: backend/app/schemas/crash.py
from __future__ import annotations

import datetime as dt
import uuid

from pydantic import BaseModel, ConfigDict


class CrashReportOut(BaseModel):
    """A stored crash, for the admin crash list/detail."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    created_at: dt.datetime
    app_version: str | None = None
    android_version: str | None = None
    device: str | None = None
    installation_id: str | None = None
    stack_trace: str | None = None
    user_comment: str | None = None


class CrashReportListOut(BaseModel):
    items: list[CrashReportOut]
    limit: int
    offset: int
    has_more: bool
